llvm/lib/IR/RepoTicket.cpp

```diff
@@ -146,8 +146,10 @@ static bool updateDigestUseDependenciesAndContributions(
            make_range(GOContributions.begin(), GOContributions.end()),
            make_range(GODependencies.begin(), GODependencies.end()))) {
     const llvm::Function *const Fn = dyn_cast<const llvm::Function>(G);
-    // if function will not be inlined, skip it
-    if (Fn && Fn->hasFnAttribute(Attribute::NoInline))
+    // if function will not be inlined and not be discarded if it is not used,
+    // skip it.
+    if (Fn && Fn->hasFnAttribute(Attribute::NoInline) &&
+        !Fn->isDiscardableIfUnused())
```

Comment:
Are there tests for both values of isDiscardableIfUnused() (true and false)?

Reply:
Yes, both cases are tested.

1. isDiscardableIfUnused() is true. 
    The new added test `repo_discardable_if_unused_test.ll` covers this case.
1. isDiscardableIfUnused() is false.
   The existing test `repo-pruning.ll` in the `llvm-project-prepo\llvm\test\Assembler` folder covers the case.